functions.py

```diff
@@ -0,0 +1,60 @@
+import requests
+from bs4 import BeautifulSoup as bs
+import json
+
+URL = "https://www.producthunt.com/"
+
+response = requests.get(URL)
+
+html = bs(response.content, 'lxml')
+
+
+tags = html.find('script', attrs = {'id':"__NEXT_DATA__"}).string.strip()
+
+
+data = json.loads(tags)
+
+
+data = data["props"]["apolloState"]
+
+ref = []  #ref will have product ids (not as it is)
+
+for i in data:
+        if "HomefeedPageFEATURED" in i:
+            for j in range(len(data[i]["items"])):
+                ref.append(data[i]["items"][j]["__ref"])
+
+upvote_list = []
+
+for k in ref:

```

Comment:
use better variable names. it would be best if you refrained from using names like I, j,e,r,l,  data, ref, etc. 
You should use a variable name such that the reader of the code understand what it hold and it type (like using dict/info/json for dictionary and list/array for lists)